### ISSUE TITLE
In functools.partial, container of subclass is not recognized as container of baseclass

### ISSUE THREAD

**Issue by RmZeta2718** [microsoft/pylance-release#4044]
## Environment data

-   Language Server version: v2023.3.10
-   OS and version: ubuntu18.04
-   Python version (& distribution if applicable, e.g. Anaconda): Anaconda python3.10 

## Code Snippet

```python
from functools import partial

class BaseClass:
    pass

class SubClass(BaseClass):
    pass

def func(base: BaseClass):
    pass

def func_list(base: list[BaseClass]):
    pass

def func_set(base: set[BaseClass]):
    pass

sub = SubClass()

func(sub)  # works fine
func_list([sub])  # works fine
func_set({sub})  # works fine

partial(func, sub)  # works fine
partial(func_list, [sub])  # error
partial(func_set, {sub})  # error
```

## Expected behavior

No errors reported.

## Actual behavior

Argument of type "list[SubClass]" cannot be assigned to parameter "base" of type "list[BaseClass]" in function func_list" Pylance(reportGeneralTypeIssues)

Argument of type "set[SubClass]" cannot be assigned to parameter "base" of type "set[BaseClass]" in function "func_set" Pylance(reportGeneralTypeIssues)

**Comment by erictraut** [comment]
Thanks for the bug report. This will be addressed in the next release.

**Comment by debonte** [comment]
This issue has been fixed in prerelease version 2023.3.21, which we've just released. You can find the changelog here: [CHANGELOG.md](https://github.com/microsoft/pylance-release/blob/main/CHANGELOG.md#2023321-8-March-2023-PreRelease)
